### PR TITLE
Bring the design section of the handbook up to date

### PR DESCRIPTION
**Pull request by kjellr**
I've been working on a rewrite + restructuring of the Design section of the Gutenberg handbook. The goals are:

- To get everything up to date. A number of our current descriptions + screenshots do not reflect the current state of Gutenberg.
- Replace most of our screenshots with more generic-looking "blueprints" so that they act more as wireframes than 1:1 design representations. This will help make sure they aren't out of date the moment something in the UI moves. Alongside that, I aimed to include more annotated visuals where I felt they'd be helpful.
- Include mobile representations of the UI alongside all of the desktop screens.
- Turn the interface blueprints into more of a comprehensive "Design Patterns" section. Add missing elements like the Block Library. 
- Update the names of interface elements to be more in line with our commonly used titles (via this [reference](https://github.com/karmatosed/talk-gutenberg-anatomyofablock/blob/master/anatomyofablock.pdf))
- Update the sketch file link to point to [SketchPress](https://github.com/10up/SketchPress). The designers are in the process of [figuring out the best approach](https://make.wordpress.org/design/2018/06/27/design-meeting-notes-for-june-27-2018/#comment-25587) for our Sketch files, but in the meantime we all love the SketchPress files. They're also updated more frequently than the ones we used to link to. 
- Also, I aimed to generally tidy up the copy just a little bit throughout.

This is a fairly comprehensive rework. The vast majority of the content that was there still exists, but has been cleaned up and moved around. 

Because this adds a lot of new content, I propose we break the "Design Principles" page out of the "Reference" section, and into its own top-level "Design" item, with 3 subpages. I've done that in this PR, but I'm 100% happy to change it. The new structure of the design content is as follows: 

- [Design](https://github.com/WordPress/gutenberg/blob/update/design-handbook/docs/design.md) (Overview, high-level principles + vision)
	- [Patterns](https://github.com/WordPress/gutenberg/blob/update/design-handbook/docs/design/design-patterns.md) (Overview of commonly used design elements + structure)
	- [Block Design](https://github.com/WordPress/gutenberg/blob/update/design-handbook/docs/design/block-design.md) (Best practices + examples for designing blocks)
	- [Resources](https://github.com/WordPress/gutenberg/blob/update/design-handbook/docs/design/design-resources.md) (Links to the SketchPress library)

_(For those looking to review, I'd suggest reading through these pages using the links above.)_

Looking forward to thoughts  + feedback.

**Still to do:**

- [x] This could definitely use a copy review. 
- [x] I'm actually not sure how the `manifest.json` works. 😄 I made a quick update in this PR, but I'd appreciate someone with more knowledge stepping in to verify that it's set up correctly.

### CODE REVIEW

**Review by karmatosed — APPROVED** [review]
I agree we should get a copy review but to me these are all great changes. This makes a lot of sense to me and thanks for taking on this big task.

File: docs/manifest.json [hunk 1]
```diff
@@ -83,6 +83,30 @@
 		"markdown_source": "https://raw.githubusercontent.com/WordPress/gutenberg/master/docs/extensibility/autocomplete.md",
 		"parent": "extensibility"
 	},
+	{
```
Comment (gziolo) [inline docs/manifest.json]: This file is autogenerated, you need to update `root-manifest.json` file instead.

**Comment by sarahmonster** [comment]
❤️ the diagrams here especially—so helpful to start developing a shared language for what we're talking about.

The new hierarchy is great, but it repeats the word "design" a whole lot:
> Design (Design Patterns, Block Design, Design Resources)

What about instead:
> Design (Patterns, Block Design, Resources)

I have some nitpicky copy suggestions. Would you prefer comments or commits? 🙂

File: docs/manifest.json (hunk 1 above)
Comment (gziolo) [reply docs/manifest.json]: Ah, once you update the proper file, you will need to run `npm run docs:build` to regenerate this file 🙃

Comment (kjellr) [reply docs/manifest.json]: 😄 Ah, that does it! Thanks @gziolo. Updated in https://github.com/WordPress/gutenberg/pull/8726/commits/03905f2735e9d3fd7a2cd926a6bdce7cee95612b

**Comment by kjellr** [comment]
@sarahmonster:

> What about instead:
> Design (Patterns, Block Design, Resources)

Sounds good to me. Updated in https://github.com/WordPress/gutenberg/pull/8726/commits/e881c559a42d7fe26cf91f6b75c4eb5d91e70122

> I have some nitpicky copy suggestions. Would you prefer comments or commits? 🙂

Feel free to commit!

File: docs/root-manifest.json [hunk 2]
```diff
@@ -83,6 +83,30 @@
...
+	},
+	{
+		"title": "Block Design",
+		"slug": "block-design",
+		"markdown_source": "https:\/\/raw.githubusercontent.com\/WordPress\/gutenberg\/master\/docs\/design\/block-design.md",
+		"parent": "design"
+	},
+	{
+		"title": "Resources",
+		"slug": "resources",
```
Comment (gziolo) [inline docs/root-manifest.json]: We already have slug resources. I'm not sure if this won't get into conflict when processed by the script which maintains Gutenberg handbook. I would recommend the following changes to ensure it works properly on wordpress.org and we can link files in both GitHub and handbook:

File renamed to `design-resources.md`.

```json
"slug": "design-resources",
"markdown_source": "https:\/\/raw.githubusercontent.com\/WordPress\/gutenberg\/master\/docs\/design\/design-resources.md",
```

File: docs/root-manifest.json [hunk 3]
```diff
@@ -83,6 +83,30 @@
...
 	},
+	{
+		"title": "Design",
+		"slug": "design",
+		"markdown_source": "https:\/\/raw.githubusercontent.com\/WordPress\/gutenberg\/master\/docs\/design.md",
+		"parent": null
+	},
+	{
+		"title": "Patterns",
+		"slug": "patterns",
```
Comment (gziolo) [inline docs/root-manifest.json]: Similar to `resources` - patterns is very short for the slug name, I think we should we do the following:

File renamed to `design-patterns.md`.

```json
"slug": "design-patterns",
"markdown_source": "https:\/\/raw.githubusercontent.com\/WordPress\/gutenberg\/master\/docs\/design\/design-patterns.md",
```

**Review by gziolo — APPROVED** [review]
Thanks for improving all the stuff related to the manifest 💯 Looks good for me on non-design related angle :)

**Comment by kjellr** [comment]
@chrisvanpatten Very cool! I could definitely see some of that content merging in here. 👍 After this merges in, let's start a new PR to discuss.

**Review by sarahmonster — APPROVED** [review]
I give this an A+. 

(Also I did some wee copyedits all over, please feel free to tweak/untweak as needed!)

**Comment by kjellr** [comment]
^ Rebased to hopefully get that check to pass...

**Comment by mizejewski** [comment]
@kjellr this is an **outstanding** re-work 👏 
I just read with fresh eyes the existing Reference > Design Principles and your proposed edits, and your version really enhances clarity. 

I like the shift to more generic-looking "blueprints" for the reasons you mention, however, it can get a little abstract (e.g. the explanation of a block chip in the Patterns section.) It might be helpful to mention/link to the more concrete examples in the Block Design section.

**Comment by gziolo** [comment]
There are 548 files changes, I don't think it's right. I'm seeing unrelated commits on the list. Something went wrong when merging changes with the master branch.

**Comment by sarahmonster** [comment]
We merged these in with a clean branch in #8924, so unless I missed anything, all these changes have been taken care of and merged into master now.

**Comment by kjellr** [comment]
@mizejewski, thanks for taking a look. (and @sarahmonster thanks for following through with the cleanup + merge!)

> It can get a little abstract (e.g. the explanation of a block chip in the Patterns section.) It might be helpful to mention/link to the more concrete examples in the Block Design section.

I think we could definitely expand some of the more potentially uncommon items (like block chips, which are based on ["Action Chips" in Material Design](https://material.io/design/components/chips.html#action-chips)), and also include some more explicit visual examples of the block design section. We'll aim to work that in to the next update.

**Pull request closed without merging**